cc/main.cc

```diff
@@ -465,46 +468,232 @@ class SelfPlayer {
   uint64_t flags_timestamp_ = 0;
 };
 
+class Evaluator {
+  class Barrier {
+   public:
+    explicit Barrier(size_t count)
```

Comment:
this barrier stuff seems somewhat complicated; is there a reason we can't just use two queues, one for each model?

Reply:
That would result in two models performing inference at the same time (with half the batch size in average). Also see my comments around line 650.